packages/block-library/src/button/index.js

```diff
@@ -32,8 +32,8 @@ const blockAttributes = {
 		attribute: 'title',
 	},
 	text: {
-		type: 'array',
-		source: 'children',
+		type: 'object',
```

Comment:
As mentioned in #9249, this is maybe a bit too easy to get wrong, and also not future proof... Maybe we either need a custom type for a `rich-text` source, or we leave the type declaration out and set it internally.